src/features/workspaces/workspace-detail/components/RoleAssignmentsTable.tsx

```diff
@@ -1,177 +1,409 @@
+            variant="compact"
+            aria-label="Group Users Table"
+            ouiaId={`${ouiaId}-users-table`}
+            columns={GROUP_USERS_COLUMNS}
+            rows={userRows}
+          />
+        </DataView>
+      </div>
+    );
   };
 
-  const rows = useMemo(
-    () =>
-      groups.map((group: Group) => ({
-        id: group.uuid,
-        row: [
-          group.name,
-          group.description ? (
-            <Tooltip isContentLeftAligned content={group.description}>
-              <span>{group.description.length > 23 ? group.description.slice(0, 20) + '...' : group.description}</span>
-            </Tooltip>
-          ) : (
-            <div className="pf-v5-u-color-400">{intl.formatMessage(messages['usersAndUserGroupsNoDescription'])}</div>
-          ),
-          group.principalCount,
-          group.roleCount,
-          '?', // not currently in API
-          group.modified ? formatDistanceToNow(new Date(group.modified), { addSuffix: true }) : '',
-        ],
-      })),
-    [groups],
-  );
-  const pageSelected = rows.length > 0 && rows.every(isSelected);
-  const pagePartiallySelected = !pageSelected && rows.some(isSelected);
-
-  const paginationComponent = (
-    <Pagination
-      perPageOptions={PER_PAGE_OPTIONS}
-      itemCount={totalCount}
-      page={page}
-      perPage={perPage}
-      onSetPage={onSetPage}
-      onPerPageSelect={onPerPageSelect}
-    />
-  );
+  const renderRolesTab = () => {
+    // Show loading state
+    if (rolesLoading) {
+      return (
+        <div className="pf-v5-u-pt-md pf-v5-u-text-align-center">
+          <Spinner size="lg" aria-label="Loading roles" />
+        </div>
+      );
+    }
 
-  return (
-    <DataView ouiaId={ouiaId} selection={selection} activeState={activeState}>
-      <DataViewToolbar
-        ouiaId={`${ouiaId}-header-toolbar`}
-        bulkSelect={
-          <BulkSelect
-            isDataPaginated
-            pageCount={groups.length}
-            selectedCount={selected.length}
-            totalCount={totalCount}
-            pageSelected={pageSelected}
-            pagePartiallySelected={pagePartiallySelected}
-            onSelect={handleBulkSelect}
+    // Show error state
+    if (rolesError) {
+      return (
+        <div className="pf-v5-u-pt-md">
+          <EmptyState variant="sm">
+            <EmptyStateHeader titleText="Unable to load roles" icon={<EmptyStateIcon icon={ExclamationCircleIcon} />} headingLevel="h4" />
+            <EmptyStateBody>{extractErrorMessage(rolesError)}</EmptyStateBody>
+          </EmptyState>
+        </div>
+      );
+    }
+
+    // Show empty state when no roles
+    if (roles.length === 0) {
+      return (
+        <div className="pf-v5-u-pt-md">
+          <EmptyState variant="sm">
+            <EmptyStateHeader titleText="No roles found" icon={<EmptyStateIcon icon={KeyIcon} />} headingLevel="h4" />
+            <EmptyStateBody>This group currently has no roles assigned to it.</EmptyStateBody>
```

Comment:
and here

Reply:
@apinkert I think you forgot to change this text and line 254 to use `intl.formatMessages`